server.js

```diff
@@ -734,13 +734,30 @@ app.post('/submit-match',  asyncMiddleware( async (req, res, next) => {
     var new_best_network_flag = false;
     var sgfbuffer = Buffer.from(req.files.sgf.data);
 
-    zlib.unzip(sgfbuffer, (err, sgfbuffer) => {
+    zlib.unzip(sgfbuffer, async (err, sgfbuffer) => {
       if (err) {
         console.error("Error decompressing sgffile in /submit-match: " + err);
       } else {
         var sgffile = sgfbuffer.toString();
         var sgfhash = checksum(sgffile, 'sha256');
 
+        var matchRequest;
+        try {
+          matchRequest = await db.collection("matchRequest").remove({
```

Comment:
I'd need to test to be sure unless you already know, if I make a joint index on these 3 fields will $in work with indexes? We won't want to do a table scan for each item we remove in the db.

Reply:
I know nothing about Mongo. Only solution i have to keep this fast is to keep this table small, rows can be flushed after a day or a few. If you put the index on random seed that should be enough to prevent a full table scan.